### PR TITLE
SLM-122: Send recipient to the create barcode endpoint

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**

### CODE REVIEW

File: integration_tests/mockApis/sendLegalMail.ts [hunk 1]
```diff
@@ -270,6 +270,9 @@ const stubCreateBarcode = (): SuperAgentRequest =>
     request: {
       method: 'POST',
       urlPattern: '/send-legal-mail/barcode',
+      bodyPatterns: [
+        { matchesJsonPath: '$[?(@.prisonerName =~ /.+/i && @.prisonId =~ /.+/i && @.contactId =~ /.+/i)]' },
```
Comment (nathanrussell-moj-digital) [inline integration_tests/mockApis/sendLegalMail.ts:274]: These 3 fields will be in the request body of the create barcode request

File: server/@types/prisonTypes/index.d.ts [hunk 2]
```diff
@@ -22,5 +22,6 @@ declare module 'prisonTypes' {
     prisonerDob?: Date
     prisonAddress: PrisonAddress
     barcodeValue?: string
+    contactId?: number
```
Comment (nathanrussell-moj-digital) [inline server/@types/prisonTypes/index.d.ts:25]: Recipient how has the contactId so we can pass it into the API

File: server/routes/barcode/contacts/CreateContactByPrisonNumberController.ts [hunk 3]
```diff
@@ -55,7 +55,13 @@ export default class CreateContactByPrisonNumberController {
     recipientForm.prisonId = createNewContactByPrisonNumberForm.prisonId
     try {
       const { prisonNumber, prisonId, prisonerName } = recipientForm
-      await this.contactService.createContact(req.session.slmToken, prisonerName, prisonId, prisonNumber)
+      const contact = await this.contactService.createContact(
```
Comment (nathanrussell-moj-digital) [inline server/routes/barcode/contacts/CreateContactByPrisonNumberController.ts:58]: When creating the Contact, we now need to capture the ID for the new contact and put it in the recipientForm.
Both controllers do this.

File: server/services/prison/PrisonRegisterService.spec.ts [hunk 4]
```diff
@@ -127,6 +127,7 @@ describe('Prison Register Service', () => {
       const prisonId = 'ASI'
 
       const expectedPrisonAddress: PrisonAddress = {
+        agencyCode: 'ASI',
```
Comment (nathanrussell-moj-digital) [inline server/services/prison/PrisonRegisterService.spec.ts:130]: Some of the tests were missing the prison `agencyCode` - these are now important as the agencyCode (prison ID) is part of the CreateBarcodeRequest body object

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**